metals/src/main/scala/scala/meta/internal/metals/FileWatcher.scala

```diff
@@ -95,7 +97,7 @@ final class FileWatcher(
       }
     }
     startWatching(new ju.ArrayList(sourceDirectoriesToWatch))
-    createdSourceDirectories.asScala.foreach { dir =>
+    createdSourceDirectories.asScala.sortBy(_.toNIO).reverse.foreach { dir =>
```

Comment:
Could you add a comment on why you need to sort and reverse it? It took me a while to realize what's going on.

Reply:
you're right. In fact we need to delete all children before parent paths.